### PR TITLE
Revamp README

### PR DESCRIPTION
**Pull request by nicoddemus**
* Use a document title
* Show a "short and sweet" section at the beginning highlighting the main usage of the plugin.
* Add a table of contents.
* Use a dedicated howto section at the end.
* Move the OVERVIEW section to the main README.

A preview can be seen here: https://github.com/nicoddemus/pytest-xdist/tree/revamp-readme

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -419,20 +397,148 @@ where the configuration file was found.
...
+
+This technique might not work for every case, but should be a starting point for many situations
+where executing a high-scope fixture exactly once is important.
+
 
-    class TestA:
-        @pytest.mark.xdist_group("group1")
-        def test2():
-            pass
+Overview
```
Comment (bluetech) [inline README.rst:455]: Maybe "How does `xdist` work?"

**Review by bluetech — APPROVED** [review]
Looks nicer to me!

File: README.rst (hunk 1 above)
Comment (nicoddemus) [reply README.rst]: Great idea, done, thanks!

**Comment by nicoddemus** [comment]
@RonnyPfannschmidt gentle ping. 😁

**Review by RonnyPfannschmidt — APPROVED** [review]

**Pull request merged**